<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QGBA::BattleChipView</class>
 <widget class="QDialog" name="QGBA::BattleChipView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>658</width>
    <height>722</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BattleChip Gate</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="1,0,0,0,0,0">
   <item>
    <widget class="QListView" name="chipList">
     <property name="acceptDrops">
      <bool>true</bool>
     </property>
     <property name="dragEnabled">
      <bool>true</bool>
     </property>
     <property name="dragDropMode">
      <enum>QAbstractItemView::DragDropMode::InternalMove</enum>
     </property>
     <property name="defaultDropAction">
      <enum>Qt::DropAction::MoveAction</enum>
     </property>
     <property name="iconSize">
      <size>
       <width>56</width>
       <height>48</height>
      </size>
     </property>
     <property name="movement">
      <enum>QListView::Movement::Free</enum>
     </property>
     <property name="isWrapping" stdset="0">
      <bool>true</bool>
     </property>
     <property name="resizeMode">
      <enum>QListView::ResizeMode::Adjust</enum>
     </property>
     <property name="gridSize">
      <size>
       <width>120</width>
       <height>72</height>
      </size>
     </property>
     <property name="viewMode">
      <enum>QListView::ViewMode::IconMode</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,0">
     <item>
      <layout class="QFormLayout" name="formLayout_4">
       <item row="0" column="1">
        <widget class="QComboBox" name="chipName"/>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Chip name</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="insert">
       <property name="text">
        <string>Insert</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QPushButton" name="save">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="load">
       <property name="text">
        <string>Load</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="add">
       <property name="text">
        <string>Add</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="remove">
       <property name="text">
        <string>Remove</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QWidget" name="advanced" native="true">
     <property name="visible">
      <bool>false</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,0,1,0,1">
      <item>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Gate type</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QRadioButton" name="gateBattleChip">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Battle Chip Gate&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Usable with &lt;span style=&quot; font-style:italic;&quot;&gt;Mega Man Battle Network 4&lt;/span&gt; and &lt;span style=&quot; font-style:italic;&quot;&gt;Rockman.EXE 4.5 Real Operation&lt;/span&gt;. &lt;/p&gt;&lt;p&gt;It can also be used to unlock minigames in &lt;span style=&quot; font-style:italic;&quot;&gt;Rockman Zero 3 &lt;/span&gt;upon inserting the &amp;quot;Meteor Red Sun&amp;quot; or &amp;quot;Blue Moon Ray&amp;quot; Battle Chip. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string notr="true">Ba&amp;ttleChip Gate</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QRadioButton" name="gateProgress">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Progress Gate&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Usable with &lt;span style=&quot; font-style:italic;&quot;&gt;Mega Man Battle Network 5&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string notr="true">Progress &amp;Gate</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QRadioButton" name="gateBeastLink">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Beast Link Gate&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Usable with &lt;span style=&quot; font-style:italic;&quot;&gt;Rockman EXE 6&lt;/span&gt;, including support for &lt;span style=&quot; font-style:italic;&quot;&gt;Mega Man Battle Network 6&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string notr="true">Beast &amp;Link Gate</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout_5">
        <item row="0" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Chip ID</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="chipId">
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="inserted">
          <property name="text">
           <string>Keep Inserted</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="updateData">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Update Chip data</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout_6">
        <item row="0" column="0" colspan="2">
         <widget class="QCheckBox" name="netGateEnable">
          <property name="toolTip">
           <string>&lt;p&gt;&lt;b&gt;Net Gate&lt;/b&gt; - Sends Battle Chip data to mGBA over TCP.&lt;/p&gt; &lt;p&gt;It can be used with an external service/app that opens a TCP socket to submit Chip IDs.&lt;/p&gt; &lt;p style=&quot;margin:0&quot;&gt;&lt;b&gt;Protocol (3 bytes)&lt;/b&gt;:&lt;/p&gt;&lt;span style=&quot;font-family:'Courier New';&quot;&gt;&lt;ol style=&quot;margin:0; padding-left:1.2em&quot;&gt; &lt;li&gt;&lt;tt&gt;0x80&lt;/tt&gt;&lt;/li&gt; &lt;li&gt;&lt;tt&gt;CHIP_ID_HI&lt;/tt&gt;&lt;/li&gt; &lt;li&gt;&lt;tt&gt;CHIP_ID_LO&lt;/tt&gt;&lt;/li&gt; &lt;/ol&gt;&lt;/span&gt;</string>
          </property>
          <property name="text">
           <string>Enable Net Gate</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="netGatePortLabel">
          <property name="text">
           <string>Local port</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="netGatePort">
          <property name="text">
           <string>2000</string>
          </property>
          <property name="maxLength">
           <number>5</number>
          </property>
          <property name="placeholderText">
           <string>1-65535</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="netGateBindLabel">
          <property name="text">
           <string>Bind address</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="netGateBind">
          <property name="text">
           <string>127.0.0.1</string>
          </property>
          <property name="placeholderText">
           <string>e.g., 127.0.0.1</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QLabel" name="netGateStatus">
          <property name="text">
           <string>Not Started</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="showAdvanced">
     <property name="text">
      <string>Show advanced</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="standardButtons">
        <set>QDialogButtonBox::StandardButton::Close|QDialogButtonBox::StandardButton::Reset</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>QGBA::BattleChipView</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>690</y>
    </hint>
    <hint type="destinationlabel">
     <x>314</x>
     <y>360</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showAdvanced</sender>
   <signal>toggled(bool)</signal>
   <receiver>advanced</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>654</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
